### PR TITLE
Prepare for MELPA

### PR DESCRIPTION
**Pull request by jcs090218**
For https://github.com/copilot-emacs/copilot.el/issues/120#issuecomment-1953697632.

### CODE REVIEW

File: copilot.el [hunk 1]
```diff
@@ -970,31 +1012,29 @@ command that triggered `post-command-hook'."
 ;; function for more information.
 (defun copilot-async-start-process (callback error-callback &rest command)
   "Start async process COMMAND with CALLBACK and ERROR-CALLBACK."
-  (let ((name (cl-first command)))
```
Comment (jcs090218) [inline copilot.el:973]: The variable name is unused.

File: copilot.el [hunk 2]
```diff
@@ -142,6 +171,68 @@ You may adjust this variable at your own risk."
...
+
+(defun copilot-turn-on-unless-buffer-read-only ()
+  "Turn on `copilot-mode' if the buffer is writable."
+  (unless buffer-read-only
+    (copilot-mode 1)))
+
+;;;###autoload
+(define-global-minor-mode global-copilot-mode
+  copilot-mode copilot-turn-on-unless-buffer-read-only)
+
```
Comment (jcs090218) [inline copilot.el:235]: Move up to avoid free reference variable warnings.

File: copilot.el [hunk 3]
```diff
@@ -142,6 +171,68 @@ You may adjust this variable at your own risk."
...
 
+;;
+;; Externals
+;;
+
+(declare-function vterm-delete-region "ext:vterm.el")
+(declare-function vterm-insert "ext:vterm.el")
+(declare-function org-sort-entries "ext:vterm.el")
+(declare-function org-entry-get "ext:vterm.el")
+(declare-function org-map-entries "ext:vterm.el")
```
Comment (emil-vdw) [inline copilot.el:182]: I don't think "ext:vterm.el" is correct here?

**Review by emil-vdw — APPROVED** [review]

File: copilot.el (hunk 3 above)
Comment (jcs090218) [reply copilot.el]: Good catch! Thank you!

**Pull request merged**